### PR TITLE
[storage] Run db-bootstrapper on local db

### PR DESCRIPTION
**Pull request by msmouse**
Introduced `DbReaderWriter` to use in place of `Arc<DbReader + DbWriter>`, because there's a catch in Rust that you can't cast `Arc<DbReader + DbWriter>` to `Arc<DbReader>`.

Used that in Executor / Db-bootstrapper interfaces, so that they can work with both local and remote DB.
 
## Motivation

It makes more sense to bootstrap the db locally instead of remotely. Plus I don't need to implement the new interfaces I needed in the storage service.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

existing coverage

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: storage/storage-interface/src/lib.rs [hunk 1]
```diff
@@ -125,6 +126,30 @@ pub trait DbWriter: Send + Sync {
...
 }
 
+#[derive(Clone)]
+pub struct DbReaderWriter {
+    pub reader: Arc<dyn DbReader>,
+    pub writer: Arc<dyn DbWriter>,
+}
+
+impl DbReaderWriter {
+    pub fn new<D: 'static + DbReader + DbWriter>(db: D) -> Self {
```
Comment (wqfish) [inline storage/storage-interface/src/lib.rs:136]: Is `'static` required by `Arc`?

Comment (msmouse) [reply storage/storage-interface/src/lib.rs:136]: Yes although I don't understand why. ☠️

**Pull request closed without merging**